pkg/apis/core/validation/validation.go

```diff
@@ -1507,12 +1539,10 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 		allErrs = append(allErrs, ValidateDNS1123Label(storageos.VolumeNamespace, fldPath.Child("volumeNamespace"))...)
 	}
 	if storageos.SecretRef != nil {
-		if len(storageos.SecretRef.Name) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "name"), ""))
-		}
-		if len(storageos.SecretRef.Namespace) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "namespace"), ""))
-		}
+		allErrs = append(allErrs, ValidateSecretReference(&core.SecretReference{
```

Comment:
I created a new core.SecreteReference object since storageos.SecretReference is of ObjectReference.
I am curious why is it declared as type ObjectReference.